PP configure-yaml updates

ceblanton opened the pull request:
Minor cleanup:
* use metomi.rose for writing rose config files
* retrieve pathnames from command line arguments instead of pp.yaml

Also simplify some of the pp.yaml component definitions, by removing timeseries and timeaverage tags that are not absolutely needed in canopy.

ceblanton commented on fre/pp/configure_script_yaml.py:
```diff
@@ -30,165 +31,94 @@ def yamlInfo(yamlfile,experiment,platform,target):
+  # disagreeable; these should be optional
+  rose_suite.set(keys=['template variables', 'DO_ANALYSIS_ONLY'],  value='False')
+  rose_suite.set(keys=['template variables', 'DO_MDTF'],  value='False')
+  rose_suite.set(keys=['template variables', 'PP_DEFAULT_XYINTERP'],  value='0,0')
```
I'm not wild about setting these defaults that really shouldn't be required, but they're currently required.

ceblanton replied:
I guess we can live with these for now.. bigger fish to worry about

ceblanton commented:
@singhd789 and @Ciheim is this update OK to go in? I'm afraid it will affect your ppp configurations but it's worth it. (I was shamelessly going to click review/merge it in but the person who submitted the PR is not allowed to review by our repo rules..)

singhd789 commented on fre/pp/ppyaml_test/pp.yaml:
```diff
@@ -1,225 +1,79 @@
-      - freq: "monthly"
-        chunkLength: *PP_AMIP_CHUNK96
-
-tmpdir:
-  tmpdirpath:
-
-install-exp-script:
-    - path: 
-    - install-option:
-        install: 
```
I wasn't sure if we still wanted these or not. Currently, they are not being used in the configure yaml script, but they are useful for running on `generic` sites (non-container use: stage-history used tmp path and the `install-option`  adds on a `symlink-dirs ...` to the cylc install command, if the user wanted the cylc-run directory to be put in a location with more space). 

Just a clarification, should we keep these for a non-container use on other sites? (and then I guess add onto the configure yaml script eventually)

ceblanton replied:
Thanks! I confess I didn't really study how these were used and had a feeling I was throwing away too much.

I'll add those back in, no problem. But I'm still not sure how they're used.. did I remove too much from configure_script_yaml.py ? Or are they used by a container specific tool?

singhd789 replied:
No, you didn't get rid of too much regarding these! The configure script didn't actually use these yet.... They were used in the old `apply-edits` script I had but when we revamped that to be configure_script_yaml, I forgot to add these back in. I can probably work on adding the functionality for these in if we want. 

They were for a non-container use. For example, if one was on niagara:
- stage-history needed a `tmp` location set (this wasn't an issue on ppan because somewhere in environment things, it was set)
- then the cylc-run directory was taking up too much space (as the default installation was `home`; disk quota exceeded), so that install option would be populated with `--symlink-dirs........` (forgot the whole syntax at the moment) to point the cylc-run directory to be installed elsewhere (`/collab1` on niagara)

ceblanton replied:
I added them back. If they're at all useful, let's keep them for sure.

ceblanton replied:
Definitely, let's keep them. I added them back in.

singhd789 commented on fre/pp/ppyaml_test/pp.yaml:
```diff
@@ -1,225 +1,79 @@
-## User defined edits
-# Paths for where the rose-suite and rose-app configurations are in checked out pp repo
```
We need to also make sure that changes here are reflected in the schema.json so the pp.yaml can be validated correctly

ceblanton replied:
Thank you, I hadn't even thought of that. The good news is I see the "yaml valid" message each time during my testing so it may be ok this case?

singhd789 replied:
True but we should definitely check. There is extra information in the schema.json now (since the updated pp.yaml would not have the other environment variables at the top)

singhd789 replied:
And we should make sure everything is defined how it's supposed to be, either required or null, yadda yadda

ceblanton replied:
You were completely right- the schema needed some corresponding updates. I think I made them now. However, I'm still not sure about the user-defined variables (the new "FRE properties").

I removed all except &xyInterp, but it's not in schema and still seems to work. Must the user variable names be in the schema? I think it would be better if they didn't.

ceblanton replied:
Thanks you're right. I removed the rose-app and suite filepath configurations from the schema

singhd789 commented on fre/pp/configure_script_yaml.py:
```diff
@@ -30,165 +31,94 @@ def yamlInfo(yamlfile,experiment,platform,target):
+          interp_split = i.get('xyInterp').split(',')
+          rose_regrid.set(keys=[f'{comp}', 'outputGridLon'], value=f'{interp_split[1]}')
+          rose_regrid.set(keys=[f'{comp}', 'outputGridLat'], value=f'{interp_split[0]}')
+          rose_regrid.set(keys=[f'{comp}', 'outputGridType'], value=f'{interp_split[0]}_{interp_split[1]}.{interp_method}')
```
Just double checking, so for the outputGridType, the output would look like this:
```outputGridType=180_288.conserve_order2```

Instead of previously, like this, for example:
```outputGridType=180_288```

Similar to `grid` comment

ceblanton replied:
Yep. those have to match until we can retire the rose app configs.. and then have those two apps use the pp.yaml directly.

singhd789 commented on fre/pp/configure_script_yaml.py:
```diff
@@ -30,165 +31,94 @@ def yamlInfo(yamlfile,experiment,platform,target):
+        # if xyInterp exists, component can be regridded
+        else:
+          interp_split = i.get('xyInterp').split(',')
+          rose_remap.set(keys=[f'{comp}', 'grid'], value=f'regrid-xy/{interp_split[0]}_{interp_split[1]}.{interp_method}')
```
Just checking for clarification and for if this is what we wanted. For grid, the output would look like this:
```grid=regrid-xy/180_288.conserve_order2```

Instead of previously, like this, for example:
```grid=regrid-xy/180_288```

ceblanton replied:
Yes, I think so. the remap files are particular to the interpolation method

singhd789 commented:
> @singhd789 and @Ciheim is this update OK to go in? I'm afraid it will affect your ppp configurations but it's worth it. (I was shamelessly going to click review/merge it in but the person who submitted the PR is not allowed to review by our repo rules..)

Hey Chris, sorry for the delay. I just added some comments

ceblanton commented:
Thanks so much for your review @singhd789 . I updated the pp schema and added back the container-specific pp.yaml items. It's ready for another review from you. No rush.. next week is fine. thanks!

ceblanton commented on fre/pp/ppyaml_test/pp.yaml:
```diff
@@ -1,225 +1,87 @@
   directories:
     history_dir: "/archive/c2b/am5/2022.01/c96L65_am5f1a0r0_amip/gfdl.ncrc5-intel22-classic-prod-openmp/history"
-    pp_dir: "/archive/$USER/canopy/am5/c96L65_amip/pp"
+    pp_dir: "/archive/$USER/canopy/am5/c96L65_amip/gfdl.ncrc5-deploy-prod-openmp/pp"
```
It's a great question. I agree with you it's not quite right yet- too much duplication.

And completely, agreed- we must document how to create a pp.yaml from scratch, with those oddies.

singhd789 commented on fre/pp/ppyaml_test/pp.yaml:
```diff
@@ -1,225 +1,87 @@
 install-exp-script:
-    - path: 
+    - path:
     - install-option:
-        install: 
+        install:
```
I'm pretty sure you didn't change anything, which was why I was confused as to why it gave me an error message, haha
